src/library/scala/collection/mutable/IndexedSeq.scala

```diff
@@ -15,4 +15,44 @@ object IndexedSeq extends SeqFactory.Delegate[IndexedSeq](ArrayBuffer)
 
 trait IndexedSeqOps[A, +CC[_], +C <: AnyRef]
   extends scala.collection.IndexedSeqOps[A, CC, C]
-  with SeqOps[A, CC, C]
+    with SeqOps[A, CC, C] {
+
+  /** Sorts this $coll in place according to an Ordering.
+    *
+    * @see [[scala.collection.SeqOps.sorted]]
+    * @param  ord the ordering to be used to compare elements.
+    * @return modified input $coll sorted according to the ordering `ord`.
+    */
+  def sortInPlace[B >: A]()(implicit ord: Ordering[B]): this.type = {
+    val len = this.length
+    if (len > 1) {
+      val arr = new Array[AnyRef](len)
+      var i = 0
+      for (x <- this) {
+        arr(i) = x.asInstanceOf[AnyRef]
+        i += 1
+      }
+      java.util.Arrays.sort(arr, ord.asInstanceOf[Ordering[Object]])
+      i = 0
+      while (i < arr.length) {
+        update(i, arr(i).asInstanceOf[A])
+        i += 1
+      }
+    }
+    this
+  }
```

Comment:
Is there a reason we can not use `Sorting.stableSort` here too?

Reply:
1. `java.util.Arrays.sort` is also used in `Seq.sorted`, I followed the pattern.
2. For `Array[AnyRef]` `Sorting.stableSort` uses `java.util.Arrays.sort`, so it actually makes sense to explicitly call it.